tests/libcxx/expected_results.txt

```diff
@@ -501,7 +501,7 @@ std/utilities/memory/allocator.traits/allocator.traits.members/destroy.pass.cpp
 std/utilities/memory/allocator.traits/allocator.traits.members/max_size.pass.cpp FAIL
 std/utilities/memory/allocator.traits/allocator.traits.members/select_on_container_copy_construction.pass.cpp FAIL
 std/utilities/memory/default.allocator/allocator.globals/eq.pass.cpp FAIL
-std/utilities/memory/default.allocator/allocator.members/allocate.pass.cpp FAIL
+std/utilities/memory/default.allocator/allocator.members/allocate.pass.cpp:1 FAIL

```

Comment:
Is it a bug that MSVC allows this test to call the non-`constexpr` functions `std::allocator::allocate` and `std::allocator::deallocate` in a constant expression?

Reply:
Hmm, seems like a bug, at least for now - I’m guessing the compiler is recognizing the Standard member function and not caring that it lacks the `constexpr` that it should have in a complete C++20 implementation. Do you want to track this somewhere? Seems like no action required to me.

Reply:
I suppose I actually meant "Is it worth filing a bug report for this?" since I can't think of a reason why anyone but us would notice or care about it. @cdacamar, do you want me to file a bug for this?

Reply:
It is standard behavior to allow a call to `std::allocator<T>::allocate` provided it meets the other requirements in `[expr.const]`: [`[expr.const]/5.19`](http://eel.is/c%2B%2Bdraft/expr.const#5.19).  The evaluation doesn't care about whether or not the member is actually `constexpr`.

Reply:
It _doesn't_ meet the other requirements in [expr.const]/5, particularly [[expr.const]/5.2](http://eel.is/c%2B%2Bdraft/expr.const#5.2). However, I think [para 6](http://eel.is/c%2B%2Bdraft/expr.const#6):

> For the purposes of determining whether an expression `E` is a core constant expression, the evaluation of a
call to a member function of `std::allocator<T>` as defined in [[allocator.members]] (http://eel.is/c%2B%2Bdraft/allocator.members), where `T` is a literal type, does not disqualify `E` from being a core constant expression, even if the actual evaluation of such a call would otherwise fail the requirements for a core constant expression.

overrules it. (Para 6 also seems to allow `std::allocator<int>{}.allocate(non_constant_expression)` to be a core constant expression - which is definitely not intended - but that's the Standard's problem, not our implementation's.)

In any case, I think I agree this isn't a bug in MSVC.

Reply:
In typical standardese fashion the context is not in the place you expect it to be 😉.